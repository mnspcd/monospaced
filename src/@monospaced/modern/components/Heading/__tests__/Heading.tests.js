/* eslint-env jest */
import { shallow } from "enzyme";

import {
  basic,
  color,
  h1,
  h2,
  h3,
  h4,
  h5,
  h6,
  level,
} from "../Heading.stories";

describe("Heading component", () => {
  it("should render correctly", () => {
    const component = shallow(basic());
    expect(component).toMatchSnapshot();
  });

  it("should render heading color correctly", () => {
    const component = shallow(color());
    expect(component).toMatchSnapshot();
  });

  it("should render heading level correctly", () => {
    const component = shallow(level());
    expect(component).toMatchSnapshot();
  });

  it("should render an H1 correctly", () => {
    const component = shallow(h1());
    expect(component).toMatchSnapshot();
  });

  it("should render an H2 correctly", () => {
    const component = shallow(h2());
    expect(component).toMatchSnapshot();
  });

  it("should render an H3 correctly", () => {
    const component = shallow(h3());
    expect(component).toMatchSnapshot();
  });

  it("should render an H4 correctly", () => {
    const component = shallow(h4());
    expect(component).toMatchSnapshot();
  });

  it("should render an H5 correctly", () => {
    const component = shallow(h5());
    expect(component).toMatchSnapshot();
  });

  it("should render an H6 correctly", () => {
    const component = shallow(h6());
    expect(component).toMatchSnapshot();
  });
});
