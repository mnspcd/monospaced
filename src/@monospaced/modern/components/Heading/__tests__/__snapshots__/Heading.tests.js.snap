// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading component should render an H1 correctly 1`] = `
<Heading
  level="1"
  size="xxxl"
>
  Hamburgefonstiv
</Heading>
`;

exports[`Heading component should render an H2 correctly 1`] = `
<Heading
  level="2"
  size="xxl"
>
  Hamburgefonstiv
</Heading>
`;

exports[`Heading component should render an H3 correctly 1`] = `
<Heading
  level="3"
  size="xl"
>
  Hamburgefonstiv
</Heading>
`;

exports[`Heading component should render an H4 correctly 1`] = `
<Heading
  level="4"
  size="l"
>
  Hamburgefonstiv
</Heading>
`;

exports[`Heading component should render an H5 correctly 1`] = `
<Heading
  level="5"
  size="m"
>
  Hamburgefonstiv
</Heading>
`;

exports[`Heading component should render an H6 correctly 1`] = `
<Heading
  level="6"
  size="s"
>
  Hamburgefonstiv
</Heading>
`;

exports[`Heading component should render correctly 1`] = `
<span
  className="Heading"
  style={
    Object {
      "color": null,
    }
  }
>
  Hamburgefonstiv
</span>
`;

exports[`Heading component should render heading color correctly 1`] = `
<span
  className="Heading Heading--xl"
  style={
    Object {
      "color": "#000a6e",
    }
  }
>
  Hamburgefonstiv
</span>
`;

exports[`Heading component should render heading level correctly 1`] = `
<h1
  className="Heading Heading--xl"
  style={
    Object {
      "color": null,
    }
  }
>
  Hamburgefonstiv
</h1>
`;
